src/gates/parse_gates.cpp

```diff
@@ -112,17 +113,31 @@ gates::CNOTAncillaPlacement determine_cnot_ancilla_placement(const std::vector<s
 
 Fraction parse_angle(std::string_view s)
 {
+    bool is_negative = false;
+    // check if negative
+    if(s.starts_with("-")) {
+        is_negative = true;
+        s = s.substr(1);
+    }
+
     if(s.starts_with("pi/"))
-        return Fraction{1,try_parse_int<ArbitraryPrecisionInteger>(s.substr(3))};
+        return Fraction{1,try_parse_int<ArbitraryPrecisionInteger>(s.substr(3)), is_negative};
 
     // use split on with *pi/ as delimiter
     auto split = lstk::split_on(s,"*pi/");
-    if(split.size() != 2)
-        throw GateParseException{lstk::cat("Could not parse angle ", s, " as n*pi/m")};
+    if(split.size() != 2) {
+        //Is this an angle of the form 3*pi?
+        // not sure why but maybe we need something like
+        std::string ns = std::string(s) + "/1";
+
+        split = lstk::split_on(ns,"*pi/");
+        if(split.size() != 2)
+            throw GateParseException{lstk::cat("Could not parse angle ", s, " as n*pi/m")};
+    }
```

Comment:
Yeah, this parser could for sure be more robust, starting from having a datatype to actually store fractions. Ideally we find a good header only library to parse and represent fractions

Reply:
there is a datatype for fractions, but was not sure how to use it for our purposes. I would not invest time in qasm parser, as i would expect somebody will eventually provide one open source on github